src/components/OpenInIgvDialog/index.tsx

```diff
@@ -183,15 +225,68 @@ const constructGDSLocalIgvUrl = async (props: { bucketOrVolume: string; pathOrKe
 
   const idx = encodeURIComponent(idxFilePresignUrl);
   const enf = encodeURIComponent(filePresignUrl);
-  const name = pathOrKey.split('/').pop() ?? pathOrKey;
-  return `http://localhost:60151/load?index=${idx}&file=${enf}&name=${name}`;
+
+  return `http://localhost:60151/load?index=${idx}&file=${enf}&name=${igvName}`;
 };
 
-const constructS3LocalIgvUrl = async (props: { bucketOrVolume: string; pathOrKey: string }) => {
-  const { bucketOrVolume, pathOrKey } = props;
+const constructS3LocalIgvUrl = (props: {
+  igvName: string;
+  bucketOrVolume: string;
+  pathOrKey: string;
+}) => {
+  const { bucketOrVolume, pathOrKey, igvName } = props;
 
-  const name = pathOrKey.split('/').pop() ?? pathOrKey;
   const file = `s3://${bucketOrVolume + '/' + pathOrKey}`;
 
-  return `http://localhost:60151/load?file=${encodeURIComponent(file)}&name=${name}`;
+  return `http://localhost:60151/load?file=${encodeURIComponent(file)}&name=${igvName}`;
+};
+
+/**
+ *
+ * We wanted to show more info in the name parameter when opening in IGV
+ * Ref: https://umccr.slack.com/archives/CP356DDCH/p1707116441928299?thread_ts=1706583808.733149&cid=CP356DDCH
+ *
+ * The desired outcome is to include libraryId, sampleId, type, and filetype
+ * Desired output: SBJ00000_L0000000_PRJ00000_tumor.bam
+ *
+ * To find the match of metadata for the specific key/path will iterate through the lims record
+ * @param props
+ */
+export const constructIgvNameParameter = ({
+  subjectData,
+  pathOrKey,
+}: {
+  pathOrKey: string;
+  subjectData: SubjectApiRes;
+}): string => {
+  const nameArray: string[] = [];
+
+  // find a match in the pathOrKey. It will return the matched key in array.
+  const findMatchingProperty = (propertyName: string) =>
+    subjectData.lims.reduce((acc, curr) => {
+      const value = curr[propertyName] as string;
+
+      // do not want value to appear twice at the return array
+      if (acc.includes(value)) {
+        return acc;
+      }
+
+      // find the matching value and push to the array
+      if (pathOrKey.toLowerCase().includes(value.toLowerCase())) {
+        acc.push(value);
+      }
+
+      return acc;
+    }, [] as Array<string>);
+
+  // 1. subjectId
+  nameArray.push(subjectData.id);
+
+  // 2 libraryId
+  nameArray.push(...findMatchingProperty('library_id'));
+
+  // 3. sampleId + filetype
+  nameArray.push(pathOrKey.split('/').pop() ?? pathOrKey);
```

Comment:
I'd like to discuss the logic a little bit here. Perhaps, we catchup tomorrow.

Especially for BAM case, we might not want to match `library_id` from the path. The path pertaining two `library_id` has other meaning as they belong to a single workflow run. 

However, a BAM file only represents 1 sample; either tumor or normal. So, we need to extract `sample_id` from the filename; strip tumor or normal (if exists); then using this `sample_id` to exchange its `library_id` counterpart from the subject meta/lims. So it is to be precise. It will be rather filter (or lookup) logic than reducing with accumulator on all possible `library_id` matches infer from the path.

What you did here is; arguable not completely wrong either - especially when we switch to VCF context. But then, that open up even more complex handling where I'd rather ignore prefix for VCF all together. Only support BAM case.

Reply:
I made the change so only bam that will have the `SBJ00000_L0000000_PRJ00000_tumor.bam` format while other filetype will only prepend the subjectId